packages/modules/viewer-react/src/hooks/useFrontstages.tsx

```diff
@@ -0,0 +1,84 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useEffect, useState } from "react";
+
+import { DefaultFrontstage } from "../components/app-ui/frontstages/DefaultFrontstage";
+import type {
+  BlankConnectionViewState,
+  ItwinViewerUi,
+  ViewerFrontstage,
+  ViewerViewCreator3dOptions,
+  ViewerViewportControlOptions,
+} from "../types";
+
+export const useFrontstages = (
+  frontstages?: ViewerFrontstage[],
+  defaultUiConfig?: ItwinViewerUi,
+  viewportOptions?: ViewerViewportControlOptions,
+  viewCreatorOptions?: ViewerViewCreator3dOptions,
+  blankConnectionViewState?: BlankConnectionViewState
+) => {
+  const [finalFrontstages, setFinalFrontstages] =
+    useState<ViewerFrontstage[]>();
+  const [noConnectionRequired, setNoConnectionRequired] =
+    useState<boolean>(false);
+  const [customDefaultFrontstage, setCustomDefaultFrontstage] =
+    useState<boolean>(false);
+
+  useEffect(() => {
+    let allFrontstages: ViewerFrontstage[] = [];
+    let defaultExists = false;
+    if (frontstages) {
+      allFrontstages = [...frontstages];
+      const defaultFrontstages = frontstages.filter(
+        (frontstage) => frontstage.default
+      );
+      if (defaultFrontstages.length > 0) {
+        setCustomDefaultFrontstage(true);
+        defaultExists = true;
+        // there should only be one, but check if any default frontstage requires an iModel connection
+        let requiresConnection = false;
+        for (let i = 0; i < defaultFrontstages.length; i++) {
+          if (defaultFrontstages[i].requiresIModelConnection) {
+            requiresConnection = true;
+            break;
+          }
+        }
+        if (!requiresConnection) {
+          // allow to continue to render
+          setNoConnectionRequired(true);
+        }
+      }
+    }
+
+    if (!noConnectionRequired) {
```

Comment:
does this get update immediately? might as well just pull `requiresConnection` up higher in scope, and check against that

Reply:
updated. I ended up changing the logic here to be simpler. We actually take a last is first approach to multiple default frontstages (see [IModelViewer.tsx](https://github.com/iTwin/viewer/blob/viewport-issue/packages/modules/viewer-react/src/components/iModel/IModelViewer.tsx#L32)) and here we were taking first is first.